### ISSUE TITLE
Must release all locks when leaving VM error

### ISSUE THREAD

**Issue by sdeleuze** [oracle/graal#2965]
On [Spring CI](https://ci.spring.io/teams/spring-graalvm-native/pipelines/spring-graalvm-native/) we see multiple build failures on Java 8 only, with both 20.3 branch and `master`, with `jvmci-20.3-b04`.

Full log: [hs_err_pid1165.log](https://github.com/oracle/graal/files/5489065/hs_err_pid1165.log)

```
# To suppress the following error report, specify this argument
# after -XX: or in .hotspotrc:  SuppressErrorAt=/javaCalls.cpp:56
#
# A fatal error has been detected by the Java Runtime Environment:
#
#  Internal Error (/b/b/e/main/graal-jvmci-8/src/share/vm/runtime/javaCalls.cpp:56), pid=1165, tid=0x00007fd81f5fe700
#  assert(!thread->owns_locks()) failed: must release all locks when leaving VM
#
# JRE version: OpenJDK Runtime Environment (8.0_272-b10) (build 1.8.0_272-b10)
# Java VM: OpenJDK 64-Bit Server VM GraalVM 20.3.0-dev (25.272-b10-jvmci-20.3-b04-fastdebug mixed mode linux-amd64 compressed oops)
# Core dump written. Default location: /home/seb/spring-graalvm-native/spring-graalvm-native-samples/webmvc-tomcat/target/native-image/core or core.1165
#
# An error report file with more information is saved as:
# /home/seb/spring-graalvm-native/spring-graalvm-native-samples/webmvc-tomcat/target/native-image/hs_err_pid1165.log
OpenJDK 64-Bit Server VM GraalVM 20.3.0-dev warning: increase O_BUFLEN in ostream.hpp -- output truncated
#
# If you would like to submit a bug report, please visit:
#   https://github.com/oracle/graal/issues
#
```

**Comment by dougxc** [comment]
Please attach `hs_err_pid7318.log` to this issue (in general you should always do this for VM crashes).

**Comment by sdeleuze** [comment]
I have [already attached the log](https://github.com/oracle/graal/files/5488801/hs_err.log) I had for 21.0, trying to get one for 20.3.

**Comment by dougxc** [comment]
This looks a lot like https://bugs.openjdk.java.net/browse/JDK-8165096 so far.

**Comment by dougxc** [comment]
> I have [already attached the log](https://github.com/oracle/graal/files/5488801/hs_err.log) I had for 21.0, trying to get one for 20.3.

Yes, sorry, I missed that upon first glance. I deleted my hasty comment but obviously not quick enough ;-)

**Comment by sdeleuze** [comment]
I have uploaded the 20.3-dev based log.

**Comment by dougxc** [comment]
This is a JDK 8 bug: https://bugs.openjdk.java.net/browse/JDK-8255910

Are you sure you've seen it occur on 11? Please attach a hs-err for that case if so.

**Comment by sdeleuze** [comment]
You are right, I confirm this bug is java 8 specific, I have updated the description accordingly.

**Comment by sdeleuze** [comment]
With `jvmci-20.3-b03` l (on Java 8) I have this error, not sure if that's the same or not.

Full log: [hs_err_pid1182.log](https://github.com/oracle/graal/files/5493254/hs_err_pid1182.log)

```
#
# A fatal error has been detected by the Java Runtime Environment:
#
#  SIGSEGV (0xb) at pc=0x00007f83768cc507, pid=1182, tid=0x00007f828a1fc700
#
# JRE version: OpenJDK Runtime Environment (8.0_272-b10) (build 1.8.0_272-b10)
# Java VM: OpenJDK 64-Bit Server VM GraalVM 20.3.0-dev (25.272-b10-jvmci-20.3-b03-fastdebug mixed mode linux-amd64 compressed oops)
# Problematic frame:
# C  [libc.so.6+0x18e507]
#
# Core dump written. Default location: /home/seb/spring-graalvm-native/spring-graalvm-native-samples/webmvc-tomcat/target/native-image/core or core.1182
#
# An error report file with more information is saved as:
# /home/seb/spring-graalvm-native/spring-graalvm-native-samples/webmvc-tomcat/target/native-image/hs_err_pid1182.log
OpenJDK 64-Bit Server VM GraalVM 20.3.0-dev warning: increase O_BUFLEN in ostream.hpp -- output truncated
[thread 140198325249792 also had an error]
#
# If you would like to submit a bug report, please visit:
#   https://github.com/oracle/graal/issues
#
Current thread is 140198639814400
Dumping core ...
Error: Image build request failed with exit status 134
```

Please let me know if I should open another issue or not.

**Comment by gilles-duboscq** [comment]
That last one looks like the same as #2164 and #2748

**Comment by sdeleuze** [comment]
Ok so I guess let's focus on the "must release all locks when leaving VM" which is currently the only one we see with b04.

**Comment by dougxc** [comment]
Tracked internally with GR-27263.

**Comment by sdeleuze** [comment]
Based on my tests this is now fixed with 20.3-b06, thanks a lot, so I am closing it. Could you please add it to 20.3 and add `spring` label please.
